ui/src/app/views/homepage/component.ts

```diff
@@ -68,13 +68,17 @@ export class HomepageComponent implements OnInit, OnDestroy {
   bugReportComponent: BugReportComponent;
   @ViewChild(QuickTourModalComponent)
   quickTourModal: QuickTourModalComponent;
+  quickTour: boolean;
 
   constructor(
     private profileService: ProfileService,
     private profileStorageService: ProfileStorageService,
     private route: ActivatedRoute,
     private router: Router,
-  ) {}
+  ) {
+    // create bound methods to use as callbacks
+    this.closeQuickTour = this.closeQuickTour.bind(this);
```

Comment:
This should happen by virtue of this being an instance method on the class. Is this actually necessary?

Reply:
I've seen the need for this in react to preserve the context of `this` when the bound method is passed down to child components - I'm interested in knowing if this is necessary too.

Reply:
Brett will be able to explain better than me, but I know it had something to do with scoping.  I can ping him for a better explanation?

Reply:
From brett: 

> In js, if you do this: `const funReference = this.fn; funReference()`, it loses the `this` binding inside the function. So if we want to pass a reference to the bound function, we have to explicitly create it. That line is effectively replacing the unbound method with a bound version, so we can pass along a reference